apps/emqx_connector/src/emqx_connector_http.erl

```diff
@@ -657,14 +636,29 @@ transform_result(Result) ->
         {error, {closed, _Message} = Reason} ->
             %% _Message = "The connection was lost."
             {error, {recoverable_error, Reason}};
-        _ ->
-            Result
+        {error, _Reason} ->
+            Result;
+        {ok, StatusCode, _} when StatusCode >= 200 andalso StatusCode < 300 ->
+            Result;
+        {ok, StatusCode, _, _} when StatusCode >= 200 andalso StatusCode < 300 ->
+            Result;
+        {ok, StatusCode, Headers} ->
+            {error, {unrecoverable_error, #{status_code => StatusCode, headers => Headers}}};
```

Comment:
Just a thought: wonder if we should treat `429 Too Many Requests` as unrecoverable.

Reply:
Good point, I think that would be an exception here.  I'll add a special clause for it.  Thanks!

Reply:
:heavy_check_mark: